### PR TITLE
support for querying pods metrics from the cluster level

### PR DESCRIPTION
**Pull request by junotx**
**What type of PR is this?**

/kind api-change

**What this PR does / why we need it**:

support for querying pods monitoring metrics from the cluster level

/area monitoring
/cc @benjaminhuo

### CODE REVIEW

**Comment by zryfish** [comment]
I believe this api can only be called by cluster administrators. @wansir Can the current authorization framework cover this API?

File: pkg/simple/client/monitoring/prometheus/promql.go [hunk 1]
```diff
@@ -349,6 +349,12 @@ func makePodMetricExpr(tmpl string, o monitoring.QueryOptions) string {
 		}
 	}
 
+	// For monitoring pods in the whole cluster
+	// Get /pods
+	if o.NamespaceName == "" && o.NodeName == "" {
+		podSelector = fmt.Sprintf(`pod=~"%s"`, o.ResourceFilter)
```
Comment (benjaminhuo) [inline pkg/simple/client/monitoring/prometheus/promql.go:355]: Please add an entry in promql_test.go for unit test

**Review by benjaminhuo — CHANGES_REQUESTED** [review]

**Comment by benjaminhuo** [comment]
> I believe this api can only be called by cluster administrators. @wansir Can the current authorization framework cover this API?

should be similar to call API for node's pods

**Review by benjaminhuo — APPROVED** [review]

**Pull request merged**